### DISCUSSION TITLE
how to update FieldSchema(vector) by FieldSchema(id,primarykey) ?

### DISCUSSION THREAD

**Discussion by studygogogo1** [milvus-io/milvus#20061] [Q&A and General discussion]
how to update FieldSchema(vector) by FieldSchema(id,primarykey) ?
I want to change vector by id . but not find API do this.

**Comment by studygogogo1** [comment]
python api

**Comment by yhmo** [comment]
Milvus doesn't support directly changing a row's values.
A workaround is:
1. delete a row by id
2. insert a new row with the id

A script example of pymilvus:
(Assume there two fields in the collection: "ID" and "Vector", now I want to change the value of row whose id is 123)
```
collection = Collection(name="xxx")
collection.delete(expr="ID=123")

newVector = [1.1, 1.2, 1.3, 1.4, ......]
data = [
        [123],
        [newVector],
    ]
collection.insert(data)
```

**Reply by Mahanchello** [reply]
Hi @yhmo, is this working? i get error: 
`pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=failed to create expr plan, expr = pk=3)>`
failing function call:
`collection.delete(expr="pk=3")`

Version: Milvus 2.2

Thank you!

**Reply by yhmo** [reply]
> Hi @yhmo, is this working? i get error: `pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=failed to create expr plan, expr = pk=3)>` failing function call: `collection.delete(expr="pk=3")`
> 
> Version: Milvus 2.2
> 
> Thank you!

The delete() interface only supports expression format like this:` pk in [1, 2, 3]`
```
expr = "pk in [3]"
collection.delete(expr=expr)
```
